docs/specification.md

```diff
@@ -587,87 +596,47 @@ Parameter (in Michelson)
 
 - Returns token pause status.
 
-**Issuing and destroying tokens**
---------------------------
+### Issuing and destroying tokens
 
 **configure_minter**
 
 Types
 ```
-configure_minter_param =
-  ( address :sender
-  , address :minter
+configure_minter =
+  ( address :minter
+  , nat     :current_minting_allowance
+  , nat     :new_minting_allowance
   )
-
-configure_minter = configure_minter_param
 ```
 
 Parameter (in Michelson)
 ```
 (pair %configure_minter
-  (address %sender)
   (address %minter)
+  (pair (nat %current_minting_allowance) (nat %new_minting_allowance))
 )
 ```
 
-- Adds `minter` to global minter list to allow him mint tokens.
+- Adds `minter` to the global minter list to allow him to mint tokens (if `minter` is not in the list already).
 
-- Sender must be master minter.
+- Sets the specified minting allowance (`new_minting_allowance`) for this minter.
 
-- Minter cannot mint tokens untill `set_minting_allowance`
-  is specified.
+- `current_minting_allowance` must be explicitly passed.
+If it does not match the actual minting allowance, the transaction MUST fail.
+It is done to prevent front-running attacks.
 
-**remove_minter**
+- Sender must be master minter.
 
-Types
-```
-remove_minter =
-  ( address :sender
-  , address :minter
-  )
-```
+**remove_minter**
 
-Parameter (in Michelson)
-```
-(pair %remove_minter
-  (address %sender)
-  (address %minter)
-)
-```
+Parameter (in Michelson): `address :minter`.
 
 - Removes minter from the global minter list and sets its minting
   allowance to 0. Once minter is removed it will no longer be able to
   mint or burn tokens.
 
 - Sender must be master minter.
 
-**set_minting_allowance**
-
-Types
-```
-set_minting_allowance_param =
-  ( address :sender
-  , address :minter
-  , nat     :amount
-  )
-
-set_minting_allowance = set_minting_allowance_param
-```
-
-Parameter (in Michelson)
-```
-(pair %set_minting_allowance
-  (address %sender)
-  (pair
-    (address %minter)
-    (nat %amount)
-))
-```
-
-- Set the amount of allowed minting allowance for an address.
-
-- Minter must present in sender's minter list.
-
 **get_minting_allowance**
```

Comment:
What is the required behavior if an address is not in the global list of minters, but actually holds some minting allowance in the ledger?

Reply:
> What is the required behavior if an address is not in the global list of minters, but actually holds some minting allowance in the ledger?

How is it possible? `removeMinter` sets minting allowance to 0, that's written in our document and in CENTRE document. But we can specify that `get_minting_allowance` returns 0 for addresses that are not minters.

Reply:
Ah yes. I somehow managed to miss that.